Sources/PackageCollections/Signature/PackageCollection+CertificatePolicy.swift

```diff
@@ -0,0 +1,31 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import struct Foundation.URL
+
+import PackageCollectionsSigning
+import TSCBasic
+
+/// Configuration in this file is intended for package collection sources to define certificate policies
+/// that are more restrictive. For example, a source may want to require that all their package
+/// collections be signed using certificate that belongs to certain subject user ID.
+internal struct PackageCollectionSourceCertificatePolicy {
+    static let sourceCertPolicies: [String: CertificatePolicyKey] = [:]
+
+    static func certificatePolicyKey(for source: Model.CollectionSource) -> CertificatePolicyKey? {
+        // Certificate policy is associated to a collection host
+        guard let host = source.url.host else { return nil }
+        return self.sourceCertPolicies[host]
+    }
+
+    /// This directory within the SwiftPM source tree contains all the root certificates required by the
+    /// certificate policies specified in this file and will be automatically trusted.
+    static let trustedRootCertsDir = AbsolutePath(#file).parentDirectory.appending(component: "RootCerts").asURL
```

Comment:
Not sure where this directory should live

Reply:
should we use somewhere under `~/.swiftpm`? we have `FileSystem.dotSwiftPM` utility and we use it package collections to store the JSON file: `fileSystem.dotSwiftPM.appending(components: "config", ...)`

Reply:
hmmm, scratch that, was reading this wrong. this should be part of the sources obviously, hence the #file

Reply:
my proposal would be to put them in `Resources/PackageCollections/RootCerts` (new top level directory named resources)

Reply:
Redoing this